website/docs/components/code-editor/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,106 @@
+
+Here is an example of a custom action to reveal secrets.
+
+![The Code Editor with the title "CodeEditor title". The custom yielded section has a "Reveal secrets" toggle.](/assets/components/code-editor/code-editor-primary-yielded-actions.png)
+
+
+!!! Warning
+
+The Code Editor has limited support for dark mode styles. Buttons have pre-defined dark mode styles, but all other components require manual color adjustments until a dark mode theme is released. Please [contact the Design Systems Team](about/support) for help translating components into dark mode.
+
+!!!
+
+## External elements
+
+Some elements or functions outside the Code Editor may affect the content within the Code Editor. In this case, we recommend turning off the header to visually couple the editor with the nearby controlling elements.
+
+![A Code Editor with the external title "Automations and expressions" coupled with a filter input, "Copy" dropdown, "Version" dropdown, and a "Create new version" button.](/assets/components/code-editor/code-editor-external-functions.png)
+
+
+## Active line highlighting
+
+When a user edits a line of code, a highlight will display to show their location within the Code Editor.
+
+![The Code Editor's line 1 with the active line highlight, shown with a blinking cursor and border around the entire line.](/assets/components/code-editor/code-editor-code-active-line.png)
+
+When a user selects a piece of code, the active line highlight no longer displays although the line number of the last line of the selection will be highlighted.
+
+![The Code Editor's line 1 to 3 selected and showing the green selection highlighted color](/assets/components/code-editor/code-editor-line-selection.png)
+
+## Bracket highlighting
+
+The Code Editor automatically closes open quotes and brackets as the user types. When the user’s cursor is next to a bracket, the bracket will highlight to show the start and end of the pair.
+
+![The Code Editor's line 1 is highlighted and the user's cursor is next to a bracket. That bracket and its pair are both highlighted with a white border.](/assets/components/code-editor/code-editor-bracket-highlighting.png)
+
+## Language
+
+Language determines how syntax highlighting is applied and formatted within the editor but is handled a bit differently between the Ember and Figma components.
+
+The **Ember** component uses [CodeMirror](https://codemirror.net/) to handle syntax highlighting and comes with a pre-defined set of languages.
+
+In **Figma** we provide a handful of example languages intended as visual examples. (Syntax highlighting in Figma is a non-trivial process and requires the manual application of color styles to each "type" of code.) You are still able to add a custom code snippet to the Figma component by typing/pasting into the text layer, but syntax highlighting will not be automatically applied.
```

Comment:
[suggestion] I don't think you need these parentheses here either, this seems like it's own sentence.

```suggestion
In **Figma** we provide a handful of example languages intended as visual examples. Syntax highlighting in Figma is a non-trivial process and requires the manual application of color styles to each "type" of code. You are still able to add a custom code snippet to the Figma component by typing/pasting into the text layer, but syntax highlighting will not be automatically applied.
```